src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -834,24 +834,33 @@ MulticopterAttitudeControl::run()
 
 	bool reset_yaw_sp = true;
 	float attitude_dt = 0.f;
-
+	static const uint8_t Max_poll_failures=3;
+	uint8_t consecutive_poll_failures=0;
+	
 	while (!should_exit()) {
 
 		poll_fds.fd = _sensor_gyro_sub[_selected_gyro];
```

Comment:
```suggestion
		sensor_correction_poll();
		poll_fds.fd = _sensor_gyro_sub[_selected_gyro];
```

Reply:
It can be an nice fix to update the selected sensor before pool. No more complications and failure counters. I agree! 
The only question is will that interfere in any way with the ACRO mode ? I.E. the code in else branch from the if(pool_fds.revents  & POLLIN). I don't see anything to be concerned about, but I just want to make sure a second pair of eyes looks at that.

Reply:
No, either way this is the safest and easiest thing to do.